src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -205,38 +205,96 @@ function updateStatsCheckBox() {
-        var i = 0;
-        for (var idx = 0; idx < panels.length; idx++) {
-            if ($(panels[idx]).attr('class').indexOf('in') === -1) {
-                setTimeout(showSingleCollapse, 50 * i, panels[idx]);
-                i++;
-            }
-        }
-        var htmlString = $(e).html();
-        htmlString = htmlString.replace(expand, collapse);
-        $(e).html(htmlString);
-        var tooltipString = $(e).attr('data-original-title').replace(expand, collapse);
-        $(e).attr('title', tooltipString).tooltip('fixTitle').tooltip('show');
+/**
+ * Expand or collapse all panels when collapse/expand panels button is clicked
+ * @param {DOM} element - The element that invoked {@code #expandOrCollapsePanels}
+ * @param {DOM} panels - The panels element to be changed.
+ */
+function expandOrCollapsePanels(element, panels) {
+    var STRING_EXPAND = 'Expand';
+    var STRING_COLLAPSE = 'Collapse';
+    // {@code panels} is undefined when the method is called by clicking on collapse panels button.
+    var panelsDefined = panels || $('div.panel-collapse');
+    var isElementAnExpandButton = $(element).html().trim().startsWith(STRING_EXPAND);
+
+    if (isElementAnExpandButton) {
+        expandPanels(panelsDefined, element);
+        replaceInHtml(element, STRING_EXPAND, STRING_COLLAPSE);
     } else {
-        var j = 0;
-        for (var k = 0; k < panels.length; k++) {
-            if ($(panels[k]).attr('class').indexOf('in') !== -1) {
-                setTimeout(hideSingleCollapse, 100 * j, panels[k]);
-                j++;
-            }
+        collapsePanels(panelsDefined);
+        replaceInHtml(element, STRING_COLLAPSE, STRING_EXPAND);
+    }
+}
+
+/**
+ * Expands all panels.
+ * @param {DOM} element - The element that was clicked to invoke {@code #expandOrCollapsePanels}
+ */
+function expandPanels(panels, element) {
+    // Expand/collapse buttons on AJAX-loaded panels have a collapse-panels-button class
+    var isAjaxLoadedPanel = $(element).is($('#collapse-panels-button'));
+    var i = 0;
+    for (var idx = 0; idx < panels.length; idx++) {
+        expandPanel(panels[idx], 50 * i, isAjaxLoadedPanel);
+        i++;
+    }
+}
+
+/**
+ * Expands {@code panel}. If the panel data is not loaded, load it by Ajax first before expansion.
+ */
+function expandPanel(panel, timeOut, isAjaxLoadedPanel) {
+    var isPanelAlreadyExpanded = $(panel).hasClass('in');
+    if (isPanelAlreadyExpanded) {
+        return;
+    }
+    if (isAjaxLoadedPanel) {
+        // When the panel's parent element has class ajax_auto or ajax-response-auto, the panel data has
+        // not been loaded yet. We will need to load the panel data by Ajax.
+        var $ajaxAuto = $(panel).parent().children('.ajax_auto');
+        var $ajaxResponseAuto = $(panel).parent().children('.ajax-response-auto');
+        var hasAjaxAutoLoading = $ajaxAuto.length !== 0;
+        var hasAjaxResponseAutoLoading = $ajaxResponseAuto.length !== 0;
```

Comment:
Name them to reflect these two cases correctly, instead of repeating the implementation detail as the variable name. After that, add comments to explain further. Basically, the reader should not have to ask the questions I'm asking.

Anyhow, may be use one variable? Replace `xxx` with a better name.

```js
var $xxx = $(panel).parent().children('.ajax_auto');
if ($xxx.length === 0) {
    $xxx = $(panel).parent().children('.ajax-response-auto');
}
```